immich-app/immich
# refactor(server): mime types

In this PR:
- Remove mime type from asset property and database table
- Remove mime type from asset response
- Remove mime-types server dependency
- Infer mime-type from file extension (on upload and retrieval)

Tested Scenarios:
- Verify `image/webp` content type for image serve (thumbnail path)
- Verify `image/jpeg` content type for image serve (resize path)
- Verify `video/mp4` content type for video serve (mp4)
- Verify `video/webm` content type for video serve (webm)
- Verify `image/jpeg` content type for file download (jpeg)
- Verify `video/mp4` content type for file download (mp4)
- Verify uploaded extensions are accepted: `.heif`, `.tiff`, `.dng`, `.ari`, `.arw`, `.raf`, `.heic`, `.mov`, `.srw`, `.pef`, `.jpg`, `.jpg`, `.mp4`, `.webm`

Next Steps:
- Add endpoint for file extension list
- Remove extension mapping from web/mobile/cli
- Remove asset type from create dto
